### ISSUE TITLE
Update from server when using mDNS fails

### ISSUE THREAD

**Issue by DropTarget** [Hieromon/AutoConnect#482]
Thank you very much for a great library! 

I am using AutoConnect for remotely deployed ESP32s and wish to add the HTTP update functionality from a remote server.  Using the "update" example provided with the library, I added mDNS registration using espmDNS as described in the documentation. All is working fine when I access AutoConnect using the WiFi.localIP address (192.168.1.122). The update proceeds normally from the remote server.  However, if I access AutoConnect using its mDNS registered hostname (actest.local), everything works except the upload file transfer.  The catalog is retrieved from the server, but when the file is selected and "update" is clicked, the GET for the selected file is never received by the file server. The update progress page is displayed but never updates.
   
I am building the firmware with platformio.  Following is the source file which is "update.ino" with espmDNS added:

*************************************************************************************************************************************

_/*
  Update.ino, Example for the AutoConnect library.
  Copyright (c) 2019, Hieromon Ikasamo
  https://github.com/Hieromon/AutoConnect
  This software is released under the MIT License.
  https://opensource.org/licenses/MIT

  This example presents the simplest OTA Updates scheme.
  When building this sketch, you may receive a compilation error message
  similar to the following:
  - text section exceeds available space in board
  This cause is the small text block size of the predetermined partition
  table. You can avoid this error by selecting Partition Scheme: from
  Arduino IDE's Tool menu and applying Minimal SPIFFS.
*/

#if defined(ARDUINO_ARCH_ESP8266)
#include <ESP8266WiFi.h>
#include <ESP8266WebServer.h>
using WebServerClass = ESP8266WebServer;
#elif defined(ARDUINO_ARCH_ESP32)
#include <WiFi.h>
#include <WebServer.h>
using WebServerClass = WebServer;
#endif
#include <AutoConnect.h>
#include <espMDNS.h>                             //<--------------------Added

// Update server setting page
static const char SETUP_PAGE[] PROGMEM = R"(
{
  "title": "Update setup",
  "uri": "/setup",
  "menu": true,
  "element": [
    {
      "name": "caption",
      "type": "ACText",
      "value": "OTA update setup",
      "style": ""
    },
    {
      "name": "isvalid",
      "type": "ACText",
      "style": "color:red"
    },
    {
      "name": "server",
      "type": "ACInput",
      "label": "Update server",
      "pattern": "^((([a-zA-Z]|[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]).)*([A-Za-z]|[A-Za-z][A-Za-z0-9-]*[A-Za-z0-9]))|((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3})$",
      "placeholder": "Your update server address"
    },
    {
      "name": "port",
      "type": "ACInput",
      "label": "port",
      "pattern": "[0-9]{1,4}"
    },
    {
      "name": "path",
      "type": "ACInput",
      "label": "path"
    },
    {
      "name": "apply",
      "type": "ACSubmit",
      "value": "Apply",
      "uri": "/apply"
    },
    {
      "name": "cancel",
      "type": "ACSubmit",
      "value": "Discard",
      "uri": "/"
    }
  ]
}
)";

// The /apply handler validates the update server settings entered on
// the setup page. APPLY_PAGE exists to enable the /apply handler,
// and its role is a page redirector. If the /apply handler detects some
// errors, the page will redirect to the /setup page with error message.
static const char APPLY_PAGE[] PROGMEM = R"(
{
  "title": "Update setup",
  "uri": "/apply",
  "menu": false,
  "element": [
    {
      "name": "redirect",
      "type": "ACElement",
      "value": "<script type=\"text/javascript\">location.href='__REDIRECT__';</script>"
    }
  ]
}
)";

WebServerClass    server;
AutoConnect       portal(server);
AutoConnectAux    setupPage;
AutoConnectAux    applyPage;
AutoConnectUpdate update;

#define UPDATESERVER_URL  ""    // Define to suit your environment 
#define UPDATESERVER_PORT 8000
#define UPDATESERVER_PATH "bin"

void loadAux() {
  setupPage.load(SETUP_PAGE);
  setupPage["server"].value = UPDATESERVER_URL;
  setupPage["port"].value = String(UPDATESERVER_PORT);
  setupPage["path"].value = UPDATESERVER_PATH;
  applyPage.load(APPLY_PAGE);
}

// The onSetup handler clears the error message field of the /setup page.
// Its field will be cleared after the /setup page generating by the
// effect of the AC_EXIT_LATER option.
String onSetup(AutoConnectAux& aux, PageArgument& arg) {
  setupPage["isvalid"].value = String();
  return String();
}

// The onApply handler validates the update server configuration.
// It does not do any semantic analysis but only verifies that the
// settings match the pattern defined in each field.
// The AutoConnectInput isValid function checks if the current value
// matches the pattern.
String onApply(AutoConnectAux& aux, PageArgument& arg) {
  String  returnUri;

  AutoConnectInput& host = setupPage["server"].as<AutoConnectInput>();
  AutoConnectInput& port = setupPage["port"].as<AutoConnectInput>();
  AutoConnectInput& path = setupPage["path"].as<AutoConnectInput>();

  Serial.printf("host: %s\n", host.value.c_str());
  Serial.printf("port: %s\n", port.value.c_str());
  Serial.printf("uri: %s\n", path.value.c_str());

  if (host.isValid() & port.isValid()) {
    update.host = host.value;
    update.port = port.value.toInt();
    update.uri = path.value;
    setupPage["isvalid"].value = String();
    returnUri = "/";
  }
  else {
    setupPage["isvalid"].value = String("Incorrect value specified.");
    returnUri = "/setup";
  }
  applyPage["redirect"].value.replace("__REDIRECT__", returnUri);
  return String();
}

void setup() {
  delay(1000);
  Serial.begin(115200);
  Serial.println();

  // Responder of root page and apply page  handled directly from WebServer class.
  server.on("/", []() {
    String content = R"(
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8" name="viewport" content="width=device-width, initial-scale=1">
</head>
<body>
Place the root page with the sketch application.&ensp;
__AC_LINK__
</body>
</html>
    )";
    content.replace("__AC_LINK__", String(AUTOCONNECT_LINK(COG_16)));
    server.send(200, "text/html", content);
  });

  // AUX page loading
  loadAux();  
  setupPage.on(onSetup, AC_EXIT_LATER);
  applyPage.on(onApply);
  portal.join({ setupPage, applyPage });
  if (portal.begin()) {                                                 //<--------------------Added if()
    update.attach(portal);
    if (MDNS.begin("actest")) {                                  //<--------------------Added
        MDNS.addService("http", "tcp", 80);                //<--------------------Added
     }                                                                           //<--------------------Added
    else                                                                       //<--------------------Added
      Serial.println("Error setting up MDNS responder");    <--------------------Added
  }                                                                               //<--------------------Added
}

void loop() 
{
   portal.handleClient();
}_

*************************************************************************************************************************************
I have turned on AC_DEBUG and the following is the output when I access with the IP address. All is working great. The update proceeds, the result is posted, and the ESP32 is restarted:
*************************************************************************************************************************************

[AC] Element<caption> not registered
[AC] caption<10> of /setup created  
[AC] *noname placed on /setup       
[AC] caption<10> of /setup loaded   
[AC] Element<isvalid> not registered
[AC] isvalid<10> of /setup created  
[AC] *noname placed on /setup
[AC] isvalid<10> of /setup loaded
[AC] Element<server> not registered
[AC] server<4> of /setup created
[AC] *noname placed on /setup
[AC] server<4> of /setup loaded
[AC] Element<port> not registered
[AC] port<4> of /setup created
[AC] *noname placed on /setup
[AC] port<4> of /setup loaded
[AC] Element<path> not registered
[AC] path<4> of /setup created
[AC] *noname placed on /setup
[AC] path<4> of /setup loaded
[AC] Element<apply> not registered
[AC] apply<9> of /setup created
[AC] *noname placed on /setup
[AC] apply<9> of /setup loaded
[AC] Element<cancel> not registered
[AC] cancel<9> of /setup created
[AC] *noname placed on /setup
[AC] cancel<9> of /setup loaded
[AC] Element<redirect> not registered
[AC] redirect<2> of /apply created
[AC] *noname placed on /apply
[AC] redirect<2> of /apply loaded
[AC] /setup on hands
[AC] /apply on hands
[AC] Set persistance
[AC] Current:saglenet
[AC] WiFi.config(IP=0.0.0.0, Gateway=0.0.0.0, Subnetmask=0.0.0.0, DNS1=0.0.0.0, DNS2=0.0.0.0)
[AC] WiFi.begin()..established IP:192.168.1.122
[AC] http server started
[AC] caption placed on /_ac/update
[AC] c1 placed on /_ac/update
[AC] firmwares placed on /_ac/update
[AC] c1 placed on /_ac/update
[AC] update placed on /_ac/update
[AC] c1 placed on /_ac/update_act
[AC] binname placed on /_ac/update_act
[AC] c2 placed on /_ac/update_act
[AC] url placed on /_ac/update_act
[AC] c3 placed on /_ac/update_act
[AC] progress_meter placed on /_ac/update_act
[AC] progress_loader placed on /_ac/update_act
[AC] c4 placed on /_ac/update_act
[AC] status placed on /_ac/update_act
[AC] c5 placed on /_ac/update_act
[AC] enable_loader placed on /_ac/update_act
[AC] c6 placed on /_ac/update_act
[AC] inprogress_meter placed on /_ac/update_act
[AC] c7 placed on /_ac/update_act
[AC] status placed on /_ac/update_result
[AC] restart placed on /_ac/update_result
[AC] /_ac/update on hands
[AC] /_ac/update_act on hands
[AC] /_ac/update_result on hands
[AC] Updater keeps callback
[AC] Element<loader> not registered
[AC] AutoConnectUpdate attached
[AC] AutoConnectUpdate enabled
[AC] Host:192.168.1.122,/_ac,generated:/_ac, allocated
[AC] Host:192.168.1.122,/_ac,already allocated
[AC] Host:192.168.1.122,/setup,generated:/setup, allocated
[AC] Host:192.168.1.122,/setup,already allocated
[AC] CB in LATER /setup
[AC] Host:192.168.1.122,/apply,generated:/apply, allocated
[AC] Host:192.168.1.122,/apply,already allocated
[AC] fetch /setup,elements stored
[AC] CB in AHEAD /apply
host: 192.168.1.101
port: 8000
uri:
[ 42064][E][WiFiClient.cpp:404] write(): fail on fd 49, errno: 104, "Connection reset by peer"
[AC] Host:192.168.1.122,/_ac,generated:/_ac, allocated
[AC] Host:192.168.1.122,/_ac,already allocated
[AC] Host:192.168.1.122,/_ac/update,generated:/_ac/update, allocated
[AC] Host:192.168.1.122,/_ac/update,already allocated
[AC] CB in AHEAD /_ac/update
[AC] Query 192.168.1.101:8000/_catalog?op=list&path=
[AC] Update server responded:
{"name":"HELOSESP32Test_firmware.bin","type":"bin","date":"04/15/22","time":"21:14:35","size":1070384}
{"name":"Simple_firmware.bin","type":"bin","date":"04/15/22","time":"09:03:48","size":1095392}
{"name":"updateserver.py","type":"file"}
{"name":"update_firmware.bin","type":"bin","date":"04/16/22","time":"20:10:28","size":1295808}
[AC] Host:192.168.1.122,/_ac/update_act,generated:/_ac/update_act, allocated
[AC] Host:192.168.1.122,/_ac/update_act,already allocated
[AC] fetch /_ac/update,elements stored
[AC] CB in AHEAD /_ac/update_act
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] 192.168.1.101:8000//update_firmware.bin update in progress...[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
 completed
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Host:192.168.1.122,/_ac/update_result,generated:/_ac/update_result, allocated
[AC] Host:192.168.1.122,/_ac/update_result,already allocated
[AC] CB in AHEAD /_ac/update_result
[AC] Host:192.168.1.122,/_ac/update_progress,ignored
[AC] Restart on update_firmware.bin updated...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:12784
load:0x40080400,len:3032
entry 0x400805e4

[AC] Element<caption> not registered
[AC] caption<10> of /setup created
[AC] *noname placed on /setup
[AC] caption<10> of /setup loaded
[AC] Element<isvalid> not registered
[AC] isvalid<10> of /setup created
[AC] *noname placed on /setup
[AC] isvalid<10> of /setup loaded
[AC] Element<server> not registered
[AC] server<4> of /setup created
[AC] *noname placed on /setup
[AC] server<4> of /setup loaded
[AC] Element<port> not registered
[AC] port<4> of /setup created
[AC] *noname placed on /setup
[AC] port<4> of /setup loaded
[AC] Element<path> not registered
[AC] path<4> of /setup created
[AC] *noname placed on /setup
[AC] path<4> of /setup loaded
[AC] Element<apply> not registered
[AC] apply<9> of /setup created
[AC] *noname placed on /setup
[AC] apply<9> of /setup loaded
[AC] Element<cancel> not registered
[AC] cancel<9> of /setup created
[AC] *noname placed on /setup
[AC] cancel<9> of /setup loaded
[AC] Element<redirect> not registered
[AC] redirect<2> of /apply created
[AC] *noname placed on /apply
[AC] redirect<2> of /apply loaded
[AC] /setup on hands
[AC] /apply on hands
[AC] Set persistance
[AC] Current:saglenet
[AC] WiFi.config(IP=0.0.0.0, Gateway=0.0.0.0, Subnetmask=0.0.0.0, DNS1=0.0.0.0, DNS2=0.0.0.0)
[AC] WiFi.begin()..established IP:192.168.1.122
[AC] http server started
[AC] caption placed on /_ac/update
[AC] c1 placed on /_ac/update
[AC] firmwares placed on /_ac/update
[AC] c1 placed on /_ac/update
[AC] update placed on /_ac/update
[AC] c1 placed on /_ac/update_act
[AC] binname placed on /_ac/update_act
[AC] c2 placed on /_ac/update_act
[AC] url placed on /_ac/update_act
[AC] c3 placed on /_ac/update_act
[AC] progress_meter placed on /_ac/update_act
[AC] progress_loader placed on /_ac/update_act
[AC] c4 placed on /_ac/update_act
[AC] status placed on /_ac/update_act
[AC] c5 placed on /_ac/update_act
[AC] enable_loader placed on /_ac/update_act
[AC] c6 placed on /_ac/update_act
[AC] inprogress_meter placed on /_ac/update_act
[AC] c7 placed on /_ac/update_act
[AC] status placed on /_ac/update_result
[AC] restart placed on /_ac/update_result
[AC] /_ac/update on hands
[AC] /_ac/update_act on hands
[AC] /_ac/update_result on hands
[AC] Updater keeps callback
[AC] Element<loader> not registered
[AC] AutoConnectUpdate attached
[AC] AutoConnectUpdate enabled

*************************************************************************************************************************************
Following is the AC_DEBUG output when Auto connect is accessed using the mDNS hostname (actest.local).  Everything works up to the point that the "GET" for the firmware file should be issued, then there is no activity.  The firmware is still alive and functions normally, but no update can be completed:
*************************************************************************************************************************************

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:12784
load:0x40080400,len:3032
entry 0x400805e4

[AC] Element<caption> not registered
[AC] caption<10> of /setup created
[AC] *noname placed on /setup
[AC] caption<10> of /setup loaded
[AC] Element<isvalid> not registered
[AC] isvalid<10> of /setup created
[AC] *noname placed on /setup
[AC] isvalid<10> of /setup loaded
[AC] Element<server> not registered
[AC] server<4> of /setup created
[AC] *noname placed on /setup
[AC] server<4> of /setup loaded
[AC] Element<port> not registered
[AC] port<4> of /setup created
[AC] *noname placed on /setup
[AC] port<4> of /setup loaded
[AC] Element<path> not registered
[AC] path<4> of /setup created
[AC] *noname placed on /setup
[AC] path<4> of /setup loaded
[AC] Element<apply> not registered
[AC] apply<9> of /setup created
[AC] *noname placed on /setup
[AC] apply<9> of /setup loaded
[AC] Element<cancel> not registered
[AC] cancel<9> of /setup created
[AC] *noname placed on /setup
[AC] cancel<9> of /setup loaded
[AC] Element<redirect> not registered
[AC] redirect<2> of /apply created
[AC] *noname placed on /apply
[AC] redirect<2> of /apply loaded
[AC] /setup on hands
[AC] /apply on hands
[AC] Set persistance
[AC] Current:saglenet
[AC] WiFi.config(IP=0.0.0.0, Gateway=0.0.0.0, Subnetmask=0.0.0.0, DNS1=0.0.0.0, DNS2=0.0.0.0)
[AC] WiFi.begin()..established IP:192.168.1.122
[AC] http server started
[AC] caption placed on /_ac/update
[AC] c1 placed on /_ac/update
[AC] firmwares placed on /_ac/update
[AC] c1 placed on /_ac/update
[AC] update placed on /_ac/update
[AC] c1 placed on /_ac/update_act
[AC] binname placed on /_ac/update_act
[AC] c2 placed on /_ac/update_act
[AC] url placed on /_ac/update_act
[AC] c3 placed on /_ac/update_act
[AC] progress_meter placed on /_ac/update_act
[AC] progress_loader placed on /_ac/update_act
[AC] c4 placed on /_ac/update_act
[AC] status placed on /_ac/update_act
[AC] c5 placed on /_ac/update_act
[AC] enable_loader placed on /_ac/update_act
[AC] c6 placed on /_ac/update_act
[AC] inprogress_meter placed on /_ac/update_act
[AC] c7 placed on /_ac/update_act
[AC] status placed on /_ac/update_result
[AC] restart placed on /_ac/update_result
[AC] /_ac/update on hands
[AC] /_ac/update_act on hands
[AC] /_ac/update_result on hands
[AC] Updater keeps callback
[AC] Element<loader> not registered
[AC] AutoConnectUpdate attached
[AC] AutoConnectUpdate enabled
[AC] Host:actest.local,/_ac,generated:/_ac, allocated
[AC] Host:actest.local,/_ac,already allocated
[AC] Host:actest.local,/setup,generated:/setup, allocated
[AC] Host:actest.local,/setup,already allocated
[AC] CB in LATER /setup
[AC] Host:actest.local,/apply,generated:/apply, allocated
[AC] Host:actest.local,/apply,already allocated
[AC] fetch /setup,elements stored
[AC] CB in AHEAD /apply
host: 192.168.1.101
port: 8000
uri:
[ 50888][E][WiFiClient.cpp:404] write(): fail on fd 49, errno: 104, "Connection reset by peer"
[AC] Host:actest.local,/_ac,generated:/_ac, allocated
[AC] Host:actest.local,/_ac,already allocated
[AC] Host:actest.local,/_ac/update,generated:/_ac/update, allocated
[AC] Host:actest.local,/_ac/update,already allocated
[AC] CB in AHEAD /_ac/update
[AC] Query 192.168.1.101:8000/_catalog?op=list&path=
[AC] Update server responded:
{"name":"HELOSESP32Test_firmware.bin","type":"bin","date":"04/15/22","time":"21:14:35","size":1070384}
{"name":"Simple_firmware.bin","type":"bin","date":"04/15/22","time":"09:03:48","size":1095392}
{"name":"updateserver.py","type":"file"}
{"name":"update_firmware.bin","type":"bin","date":"04/16/22","time":"20:10:28","size":1295808}
[AC] Host:actest.local,/_ac/update_act,generated:/_ac/update_act, allocated
[AC] Host:actest.local,/_ac/update_act,already allocated
[AC] fetch /_ac/update,elements stored
[AC] CB in AHEAD /_ac/update_act

*************No further activity *****************
*************************************************************************************************************************************
Thinking this might be an mDNS issue, I also tried to use the arduino_mdns library, but the results were the same.  Please let me know if I can provide any additional information.

Thanks,
DropTarget

**Comment by DropTarget** [comment]
Additional information:
All above tests were performed by accessing AutoConnect from Windows 10 using Firefox browser.  I just tried it on my Windows 11 laptop with Firefox and it works fine.  This leads me to believe there might be an MDNS issue with Windows 10.

**Comment by Hieromon** [comment]
@DropTarget Having said that, I too will try with the sketch code you posted. I may get different results in my environment. I'll keep you posted if I find out anything.

**Comment by DropTarget** [comment]
@Hieromon
I opened port 5353 for UDP. I am able to resolve actest.local via powershell regardless of any wait time between tries.  It doesn't appear that ESP32 requires periodic update of MDNS.
Update result hang is the same on Windows 10 with port 5353 open. I have added additional AC_DEBUG statements in an effort to home in on where the behavioral difference is, but have not had any success yet.  I might need to set up JTAG debug to get to the root cause.

**Comment by Hieromon** [comment]
@DropTarget I feel like this is not a cause with the Windows 10 mDNS responder and still think something happens on the ESP32 side.
I noticed in your log that it includes a connection reset by peer error. If this is the cause, I believe the XHR from AutoConnectUpdate is pending.

**Comment by DropTarget** [comment]
@Hieromon I turned on the inspector and never saw any pending packets.  When using MDNS, it never makes it past the first update progress as seen here:
![image](https://user-images.githubusercontent.com/17154098/165669441-57df7088-6986-4c5a-8a6f-66d94e31b262.png)

I have attached two wireshark captures, success and fail. In these traces, the nodes are:
192.168.1.113 Windows 10 Web client
192.168.1.124 Update server running updateserver.py -b 192.168.1.124
192.168.1.125 ESP32 running the test firmware 

In the success trace, note that just before the transfer starts at packet #757 is an ARP request from the web client requesting the esp32 IP address.  This is followed immediately by a response, and update progress, and the data transfer.

In the fail trace, at packet #842, is the corresponding ARP request but no response.  This is the last we hear from the web client or the update server.
 
Do you have any idea why the esp32 isn't responding to the ARP request?

[wireshark.zip](https://github.com/Hieromon/AutoConnect/files/8579288/wireshark.zip)

**Comment by DropTarget** [comment]
@Hieromon I added WiFi.setSleep(false); immediately after successful return from Portal.begin but it made no difference. Please let me know if I can supply any additional information.

**Comment by DropTarget** [comment]
@Hieromon 
I've not noticed any other anomalies in the network that would lead to a clue on this issue. It appears that I have one device in the network with a class A address.  It's an iRobot Roomba that connects to the cloud.  If I get a chance, I'll disconnect it from the network, reset the gateway, and try again. I confirmed again today that this doesn't seem to be an issue with Windows 11 acting as web client. This could simply be due to timing or other factors not causing the same set of conditions.  I also confirmed successful update operation with Ubuntu Server 20.0.4 acting as web client.  If I get any additional information, I'll let you know.  In the meantime, we can close this issue since I have workarounds.

Thanks for your help and a fantastic library.
